src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -899,8 +899,19 @@ public override AwaitExpressionInfo GetAwaitExpressionInfo(AwaitExpressionSyntax
                 throw new ArgumentException("node.Kind==" + node.Kind());
             }
 
-            var bound = GetUpperBoundNode(node);
-            BoundAwaitableInfo awaitableInfo = (((bound as BoundExpressionStatement)?.Expression ?? bound) as BoundAwaitExpression)?.AwaitableInfo;
+            //usually the GetUpperBoundNode is sufficient, but we need to handle
+            //for example BoundExpressionStatement or BoundConversion here, too.
+            BoundAwaitExpression boundAwait = null;
+            foreach (var boundNode in GetBoundNodes(node))
```

Comment:
> foreach (var boundNode in GetBoundNodes(node)) [](http://example.com/codeflow?start=11&length=47)

We should not be inventing new ways to locate nodes. If ```GetLowerBoundNode``` isn't working, I would like to understand why. Otherwise, we should simply use it because it appears that the bug was caused by the usage of the wrong API.

Reply:
Okay, I use simply `GetLowerBoundNode`